core/src/labels/labels.cpp

```diff
@@ -104,6 +105,54 @@ void Labels::update(const View& _view, float _dt, const std::vector<std::unique_
         }
     }
 
+    /// Mark labels to skip transitions
+
+    for (const auto& t0 : _tiles) {
```

Comment:
Shouldn't the transition state be only checked when _view.getNextZoom() != currentZoom? 

Hmm, still this only works for a non-tilted view: I think we could also handle the case where zoom-level changes of tiles are not equivalent to zoom-level changes of the view. I.e when proxy tiles were temporarily rendered and for tilted views. 
To do this we need to add a way to determine the first time when a tile is rendered and check the label state of the previously active tile(s) for its area.